connector/connect/server/src/main/scala/org/apache/spark/sql/connect/utils/ErrorUtils.scala

```diff
@@ -213,6 +221,30 @@ private[connect] object ErrorUtils extends Logging {
       .exists(_.toString.contains("org.apache.spark.sql.execution.python"))
   }
 
+  def handleNonQueryError[V](
+      opName: String,
+      observer: StreamObserver[V],
+      sessionHolderOpt: Option[SessionHolder],
+      callback: Option[() => Unit] = None): PartialFunction[Throwable, Unit] = {
+    val partial: PartialFunction[Throwable, Throwable] = {
+      case e: Throwable if e.isInstanceOf[SparkThrowable] || NonFatal.apply(e) =>
+        StatusProto.toStatusRuntimeException(buildStatusFromThrowable(e, sessionHolderOpt))
+      case e: StatusRuntimeException => e
```

Comment:
this case is not present in the query `handleError`, why it's a case needed here?

Reply:
In the original query error case, I was very certain that the error conversion does not happen inside the try block. Let me research why I added this (or if I was just cautious)